Formula/y/yass-cli.rb

```diff
@@ -0,0 +1,70 @@
+class YassCli < Formula
+  desc "Lightweight and efficient, socks/http forward proxy"
+  homepage "https://letshack.info"
+  url "https://github.com/Chilledheart/yass/releases/download/1.13.1/yass-1.13.1.tar.bz2"
+  sha256 "26015d32a6bdc5bf2670b36c8f5fb68f4a324d577e027bc5eef18ae977e00c9f"
+  license "GPL-2.0-only"
+  head "https://github.com/Chilledheart/yass.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "go" => :build
+  depends_on "llvm" => :build if DevelopmentTools.clang_build_version < 1500
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "c-ares"
+  depends_on "libnghttp2"
+  depends_on "mbedtls"
+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "zlib"
+  end
+
+  def install
+    ENV.llvm_clang if OS.mac? && (DevelopmentTools.clang_build_version < 1500)
+
+    args = %w[
+      -DBUILD_SHARED_LIBS=off
```

Comment:
```suggestion
      -DBUILD_SHARED_LIBS=on
```

Reply:
It is required by yass project and we're unlikely to remove it.

Reply:
Why is it required?

Reply:
For example, it compiles an upstream libc++ library and is linked statically into yass because macOS below 12.0 doesn't have the correct c++20 std library. It will cause troubles if it is linked shared library. There are more than 10 static libraries (mostly, c++ libraries) inside yass project and untested for shared linkage. We just avoid to run into dependency hell.

Reply:
But we're not building on versions of macOS below macOS 12, so we shouldn't be using a custom libc++.

Reply:
Okay. It breaks on macOS 13.0 when building with system libc++ because missing features of std::filesystem (I tried system libc++ at first commit).

Reply:
Use the `libc++` from `llvm` like this:

https://github.com/Homebrew/homebrew-core/blob/5ffbd6b840b1a95da482f8b2a10729e76628d749/Formula/d/dwarfs.rb#L80-L86

You'll need to drop `=> :build` from the `llvm` dependency.

Reply:
It is too heavy to depends on versioned llvm even if appleclang is equal or more than 1500. At least we use the upstream libc++ which is the same revision used in production branch of chrome and chromium which is well tested on macOS. 

We don't get the same guarantee from versioned release of llvm.

Reply:
It's still gated by `if DevelopmentTools.clang_build_version < 1500` so it won't be a dependency on newer versions of macOS. The libc++ from the versioned release of llvm is also well-tested on macOS in our CI and by our users.

Reply:
I still don't think additional 1.8G disk usage is a good option.
<img width="651" alt="截屏2024-09-08 20 19 02" src="https://github.com/user-attachments/assets/736c45ae-d834-4501-b93d-c78c25dfb627">

yass's bundled libc++ is well tested as well and we don't use llvm on macOS 13.0 because the appleclang is 1500.

Reply:
<img width="976" alt="截屏2024-09-08 20 21 57" src="https://github.com/user-attachments/assets/7ddab833-8efb-4df8-a544-1b947c1e55a3">

screenshot from https://ci.chromium.org/p/chromium/g/main/console.

Reply:
<img width="607" alt="截屏2024-09-08 20 24 09" src="https://github.com/user-attachments/assets/458a5d3a-63b4-413c-9ef6-10eb09b99c6f">

<img width="762" alt="截屏2024-09-08 20 26 08" src="https://github.com/user-attachments/assets/4ebf717f-25ef-4682-833e-2456da83f2eb">

Reply:
Disk space is cheap, and using a custom libc++ goes against https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

Reply:
> we don't use llvm on macOS 13.0 because the appleclang is 1500 there.

Reply:
system libc++ is cheap if it doesn't implement the std ts. Even it is possible to like static libc++ library in llvm recipe. We are unlikely to add custom libc++ support (for yass, llvm recipe's builtin libc++ library is custom one)

Reply:
<img width="1250" alt="截屏2024-09-08 20 41 04" src="https://github.com/user-attachments/assets/638e6ee0-b9a9-4baa-b36e-7dbfc00cdb5b">

For macOS 13.0 build, there is no llvm packages. I don't think it is a good way to keep use system libc++ in newer macOS and use custom libc++ for  old macOS. It is burden to homebrew rules as you described as well.


https://github.com/Homebrew/homebrew-core/actions/runs/10759695738/job/29836767807?pr=183910

Reply:
And we always find bugs in old libc++ which is not fixed and a burden to make workaround in yass side.

Reply:
> For macOS 13.0 build, there is no llvm packages.

I don't know what this means.

> I don't think it is a good way to keep use system libc++ in newer macOS and use custom libc++ for old macOS.

I disagree. We should use system libc++ where we can, and LLVM libc++ where we can't.

> It is burden to homebrew rules as you described as well.

It is not.